### PR TITLE
feat: Implemented post-deployment validation

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Implemented post-deployment validation
- Improved robustness of `diff` logic

| Pipeline | 
| - |
| https://github.com/AlexanderSehr/bicep-registry-modules/actions/runs/6982605270/job/19002095467#step:4:165 |

## Example Output
```
VERBOSE: Invoke function with
  VERBOSE: {
    "Version": "0.3",
    "ModulePath": "avm/res/key-vault/vault"
  }
  
  VERBOSE: Requested HTTP/1.1 GET with 0-byte payload
  VERBOSE: Received HTTP/1.1 71191-byte response of content type application/json
  VERBOSE: Bicep modules found in MCR catalog:
  
  bicep/samples/hello-world
  bicep/samples/array-loop
  bicep/ai/bing-resource
  bicep/ai/cognitiveservices
  bicep/app/app-configuration
  bicep/app/dapr-containerapp
  bicep/app/dapr-containerapps-environment
  bicep/authorization/resource-scope-role-assignment
  bicep/azure-gaming/game-dev-vm
  bicep/azure-gaming/game-dev-vmss
  bicep/compute/availability-set
  bicep/compute/container-registry
  bicep/compute/custom-image-vmss
  bicep/compute/event-hub
  bicep/compute/function-app
  bicep/cost/resourcegroup-scheduled-action
  bicep/cost/subscription-scheduled-action
  bicep/deployment-scripts/aks-run-command
  bicep/deployment-scripts/aks-run-helm
  bicep/deployment-scripts/build-acr
  bicep/deployment-scripts/create-kv-certificate
  bicep/deployment-scripts/create-kv-sshkeypair
  bicep/deployment-scripts/import-acr
  bicep/deployment-scripts/wait
  bicep/lz/sub-vending
  bicep/identity/user-assigned-identity
  bicep/network/virtual-network
  bicep/network/traffic-manager
  bicep/network/nat-gateway
  bicep/network/dns-zone
  bicep/network/public-ip-address
  bicep/network/public-ip-prefix
  bicep/network/private-dns-zone
  bicep/observability/grafana
  bicep/search/search-service
  bicep/security/keyvault
  bicep/storage/cosmos-db
  bicep/storage/data-explorer
  bicep/storage/storage-account
  bicep/storage/log-analytics-workspace
  bicep/storage/postgresql-single-server
  bicep/storage/redis-cache
  bicep/storage/mysql-single-server
  bicep/avm/res/batch/batch-account
  bicep/avm/res/cognitive-services/account
  bicep/avm/res/compute/ssh-public-key
  bicep/avm/res/event-grid/domain
  bicep/avm/res/event-grid/system-topic
  bicep/avm/res/insights/action-group
  bicep/avm/res/insights/component
  bicep/avm/res/insights/diagnostic-setting
  bicep/avm/res/key-vault/vault
  bicep/avm/res/kubernetes-configuration/extension
  bicep/avm/res/kubernetes-configuration/flux-configuration
  bicep/avm/res/logic/workflow
  bicep/avm/res/network/dns-forwarding-ruleset
  bicep/avm/res/network/dns-resolver
  bicep/avm/res/network/dns-zone
  bicep/avm/res/network/express-route-circuit
  bicep/avm/res/network/express-route-gateway
  bicep/avm/res/network/load-balancer
  bicep/avm/res/network/network-interface
  bicep/avm/res/network/private-dns-zone
  bicep/avm/res/network/private-endpoint
  bicep/avm/res/network/public-ip-address
  bicep/avm/res/operational-insights/workspace
  bicep/avm/res/operations-management/solution
  bicep/avm/res/power-bi-dedicated/capacity
  bicep/avm/res/resource-graph/query
  bicep/avm/res/search/search-service
  bicep/avm/res/sql/server
  
  
  VERBOSE: Passed: Found module [avm/res/key-vault/vault] in the MCR catalog
  VERBOSE: Requested HTTP/1.1 GET with 0-byte payload
  VERBOSE: Received HTTP/1.1 87-byte response of content type application/json
  VERBOSE: Tags for module in path [avm/res/key-vault/vault] found in MCR catalog:
  0.2.0
  0.3.0
  
  Passed: Found new tag [0.3] for published module
```

### CODE REVIEW

**Review by skydiver1980 — APPROVED** [review]

**Review by skydiver1980 — APPROVED** [review]

**Review by ChrisSidebotham — DISMISSED** [review]
🤌

File: avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            Start-Sleep -Seconds $retry_seconds
+        }
+
+        if ((Get-Date) -ge $end_time) {
+            throw "Time limit reached. Failed to validate publish of module in path [$ModulePath] within the specified time."
+        }
+    }
+
+    while ($true) {
+        $existingTagsUrl = "https://mcr.microsoft.com/v2/bicep/$ModulePath/tags/list"
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1:55]: Re-the url

File: avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1 [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<#
+.SYNOPSIS
+Check if a module in a given path is published in a given version
+
+.DESCRIPTION
+Check if a module in a given path is published in a given version. Tries to find the module for a maximum of 6 minutes.
```
Comment (eriqua) [inline avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1:6]: So should we search for 6 minutes or 1 hour?

Comment (AlexanderSehr) [reply avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1:6]: Both. The one 1h is contained in the 6 minutes.

File: avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1 [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+.PARAMETER Version
+Mandatory. The version of the module to check for. For example: '0.2.0'
+
+.PARAMETER ModulePath
```
[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/utilities/pipelines/publish/Confirm-ModuleIsPublished.ps1]: Changed to `PublishedModuleName` as per preceeding publish script the value is coming from.

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**